engine/installation/linux/ubuntulinux.md

```diff
@@ -116,365 +119,494 @@ packages from the Docker repository:
-    kernel does not support swap limit capabilities. Limitation discarded.
+```none
+WARNING: Your kernel does not support cgroup swap limit. WARNING: Your
+kernel does not support swap limit capabilities. Limitation discarded.
+```
 
-To prevent these messages, enable memory and swap accounting on your
-system.  Enabling memory and swap accounting does induce both a memory
-overhead and a performance degradation even when Docker is not in
-use. The memory overhead is about 1% of the total available
-memory. The performance degradation is roughly 10%.
+If you don't care about these capabilities, you can ignore the warning. You can
+enable these capabilities in your kernel by following these instructions. Memory
+and swap accounting incur an overhead of about 1% of the total available
+memory and a 10% overall performance degradation, even if Docker is not running.
 
-To enable memory and swap on system using GNU GRUB (GNU GRand Unified
-Bootloader), do the following:
+1.  Log into Ubuntu as a user with `sudo` privileges.
 
-1. Log into Ubuntu as a user with `sudo` privileges.
+2.  Edit the `/etc/default/grub` file.
 
-2. Edit the `/etc/default/grub` file.
+3.  Add or edit the `GRUB_CMDLINE_LINUX` line to add the following two key-value
+    pairs:
 
-3. Set the `GRUB_CMDLINE_LINUX` value as follows:
+    ```none
+    GRUB_CMDLINE_LINUX="cgroup_enable=memory swapaccount=1"
+    ```
 
-        GRUB_CMDLINE_LINUX="cgroup_enable=memory swapaccount=1"
+    Save and close the file.
 
-4. Save and close the file.
+4.  Update GRUB.
 
-5. Update GRUB.
+    ```bash
+    $ sudo update-grub
+    ```
 
-        $ sudo update-grub
+     If your GRUB configuration file has incorrect syntax, an error will occur.
+     In this case, steps 3 and 4.
 
-6. Reboot your system.
+6.  Reboot your system. Memory and swap accounting are enabled and the warning
+    does not occur.
 
 
 ### Enable UFW forwarding
 
 If you use [UFW (Uncomplicated Firewall)](https://help.ubuntu.com/community/UFW)
 on the same host as you run Docker, you'll need to do additional configuration.
 Docker uses a bridge to manage container networking. By default, UFW drops all
-forwarding traffic. As a result, for Docker to run when UFW is
-enabled, you must set UFW's forwarding policy appropriately.
+forwarding traffic. You must set UFW's forwarding policy appropriately.
 
-Also, UFW's default set of rules denies all incoming traffic. If you want to
-reach your containers from another host allow incoming connections on the Docker
-port. The Docker port defaults to `2376` if TLS is enabled or `2375` when it is
-not. If TLS is not enabled, communication is unencrypted. By default, Docker
-runs without TLS enabled.
+In addition, UFW blocks all incoming traffic by default. If you want to access
+your containers from another host, you need to configure UFW to allow incoming
+connections on the Docker port, which defaults to `2376` if TLS encrypted
```

Comment:
Looks like some things are mashed together here (already in the original text);

- ports 2376 or 2375 are not needed to access your containers, and only used to remotely access the Docker Remote API. By default, remote access is disabled (the daemon only listens on a socket), so unless reconfigured, these ports don't have to be opened.
- yes, port 2375 (non-TLS) can be used for the remote API, but that's the _biggest_ "no no", as not having it protected by TLS means giving anyone root access to your machine (without logging in, or any security whatsoever) - possibly a warning should be repeated here as well (wdyt?)